lib/TypeNames/ModelToPTMLTypeHelpers.cpp

```diff
@@ -295,13 +295,33 @@ static void printDefinition(const model::EnumDefinition &E,
   Header << std::move(Suffix) << ";\n";
 }
 
+static void printPadding(ptml::PTMLIndentedOstream &Header,
+                         ptml::PTMLCBuilder &B,
+                         uint64_t FieldOffset,
+                         uint64_t NextOffset,
+                         bool ForEditing) {
+  revng_assert(FieldOffset <= NextOffset);
+  if (FieldOffset == NextOffset)
+    return; // No padding is needed
+
+  if (ForEditing) {
```

Comment:
`ForEditing` is not great, there should be a boolean for each feature (e.g., `UseStartAt`), but at the current stage this would require rolling back the existing usages of `ForEditing` and probably exploding them in multiple arguments.

@pfez I'd suggest to let @Cvelth refactor this file so that we something more human like:

```c++
class Printer {
private:
  ptml::PTMLIndentedOstream &Header;
  ptml::PTMLCBuilder &B;
  bool UseStartAt;
  bool SomeOtherFeatureThatCurrentlyIsEmbeededInForEditing;
  Logger<> &Log;
  // ...

public:
  void printPadding(uint64_t FieldOffset, uint64_t NextOffset);
  void printDefinition(const model::StructDefinition &S);
  // ...
}
```

This is something we've been postponing way too long and it's starting to prevent us from having good designs.

Reply:
I agree

Reply:
Sounds good!

Reply:
Nice, btw I'd be fine if this was an extra commit at the end.

Reply:
Please take a look at #31 as an a bit of an overkill-y extension of this idea.

(I'm closing this thread, feel free to reopen it if you have further comments)